### DISCUSSION TITLE
When using wire:navigate, livewire will clear the data in Alpine.store without re-initializing it, resulting in the error "Uncaught TypeError: Cannot redefine property: $nextTick"

### DISCUSSION THREAD

**Discussion by xiaoxidashen** [livewire/livewire#7996] [7. Bugs (archived)] [answered]
### Livewire version

3.4.4

### Laravel version

10.43.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

use @include('components.global-js') in layout blade file,
and put follow code into resources/views/components/global-js.blade.php
bug description in code comment
```php
{{--resources/views/components/global-js.blade.php--}}
<div x-data="$store.somedata" x-show="show">I am hidden</div>

<script>
    function alpine_init() {
        Alpine.store('somedata', {
            show: false,
        });
        Alpine.store('init', 'true');
    }

    document.addEventListener('alpine:init', () => {
        alpine_init();
    });

    // If you do not add the following code, when you use wire:navigate, the following error will be reported:
    // livewire.js?id=a27c4ca2:1217  Uncaught TypeError: Cannot redefine property: $nextTick
    // at Function.defineProperty (<anonymous>)
    //     at livewire.js?id=a27c4ca2:1217:14
    //     at Array.forEach (<anonymous>)
    //         at injectMagics (livewire.js?id=a27c4ca2:1205:28)
    //         at Function.<anonymous> (livewire.js?id=a27c4ca2:3505:5)
    //             at flushHandlers (livewire.js?id=a27c4ca2:1390:48)
    //             at stopDeferring (livewire.js?id=a27c4ca2:1395:7)
    //             at deferHandlingDirectives (livewire.js?id=a27c4ca2:1398:5)
    //             at Object.initTree (livewire.js?id=a27c4ca2:913:5)
    //             at nowInitializeAlpineOnTheNewPage (livewire.js?id=a27c4ca2:7644:13)
    //             (anonymous) @ livewire.js?id=a27c4ca2:1217
    //             injectMagics @ livewire.js?id=a27c4ca2:1205
    //             (anonymous) @ livewire.js?id=a27c4ca2:3505
    //             flushHandlers @ livewire.js?id=a27c4ca2:1390
    //             stopDeferring @ livewire.js?id=a27c4ca2:1395
    //             deferHandlingDirectives @ livewire.js?id=a27c4ca2:1398
    //             initTree @ livewire.js?id=a27c4ca2:913
    //             nowInitializeAlpineOnTheNewPage @ livewire.js?id=a27c4ca2:7644
    //             (anonymous) @ livewire.js?id=a27c4ca2:7595
    //             (anonymous) @ livewire.js?id=a27c4ca2:7636
    try {
        if (Alpine && Alpine.store && !!Alpine.store('init')) {
            alpine_init();
        }
    } catch (e) {}
</script>

```

### Repository or Wirebox URL

no Repository 

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by danie-ramdhani** [comment]
why not use `@script` (https://livewire.laravel.com/docs/javascript#executing-scripts)?

```blade
@script
<script>
    Alpine.store('somedata', {
        show: false,
    });
    Alpine.store('init', 'true');
</script>
@endscript
```

**Reply by xiaoxidashen** [reply]
Thank you for your answer. I tried your solution. I used `php artisan make:volt global-volt --class` to create a page, and then replaced @include('components.global-js') globally with <livewire :global-volt/>, and now no error is reported when using wire:navigate to move forward, but when I use the browser's back button to return, an error is reported as follows,But when I use my original solution, no error is reported.
```
Uncaught TypeError: Cannot redefine property: $nextTick
    at Function.defineProperty (<anonymous>)
    at livewire.js?id=a27c4ca2:1217:14
    at Array.forEach (<anonymous>)
    at injectMagics (livewire.js?id=a27c4ca2:1205:28)
    at Function.<anonymous> (livewire.js?id=a27c4ca2:3505:5)
    at flushHandlers (livewire.js?id=a27c4ca2:1390:48)
    at stopDeferring (livewire.js?id=a27c4ca2:1395:7)
    at deferHandlingDirectives (livewire.js?id=a27c4ca2:1398:5)
    at Object.initTree (livewire.js?id=a27c4ca2:913:5)
    at nowInitializeAlpineOnTheNewPage (livewire.js?id=a27c4ca2:7644:13)
(anonymous) @ livewire.js?id=a27c4ca2:1217
injectMagics @ livewire.js?id=a27c4ca2:1205
(anonymous) @ livewire.js?id=a27c4ca2:3505
flushHandlers @ livewire.js?id=a27c4ca2:1390
stopDeferring @ livewire.js?id=a27c4ca2:1395
deferHandlingDirectives @ livewire.js?id=a27c4ca2:1398
initTree @ livewire.js?id=a27c4ca2:913
nowInitializeAlpineOnTheNewPage @ livewire.js?id=a27c4ca2:7644
(anonymous) @ livewire.js?id=a27c4ca2:7617
(anonymous) @ livewire.js?id=a27c4ca2:7636
```

resources\views\livewire\global-volt.blade.php 

```php
<?php

use Livewire\Volt\Component;

new class extends Component {
    //
}; ?>

{{--resources/views/livewire/global-volt.blade.php--}}
<div x-data="$store.somedata" x-show="show">I am hidden</div>

@script
<script>
    Alpine.store('somedata', {
        show: false,
    });
    Alpine.store('init', 'true');
</script>
@endscript
```

**Comment by danie-ramdhani** [comment]
here made the example based on your code:

```blade
    <div x-show="$store.somedata.show" x-cloak>I am hidden</div>

    <button type="button" x-on:click="$store.somedata.show = !$store.somedata.show">
        Toggle
    </button>

    @script
        <script>
            Alpine.store('somedata', {
                show: false,
            });
        </script>
    @endscript
```

you need to learn how alpine js store works first (https://alpinejs.dev/globals/alpine-store).

**Reply by danie-ramdhani** [reply, marked as answer]
oh, one more thing. please wrap all elements in a single root element (div element):

```blade
<div>
 <div x-show="$store.somedata.show" x-cloak>I am hidden</div>

    <button type="button" x-on:click="$store.somedata.show = !$store.somedata.show">
        Toggle
    </button>
</div>
```

**Reply by xiaoxidashen** [reply]
Of course, thank you for your reply. I tried your solution and it worked perfectly. I now know that the key to the problem is to use something like `<div x-show="$store.somedata.show">it is working</div> `, instead of `<div x-data="$store.somedata" x-show="show">not working</div>`, I saw the tutorial link you sent, but the tutorial did not explain why it cannot be used. Similar to `x-data="$store.somedata" x-show="show"`, except that the usage plan for demonstration is similar to `<div x-data :class="$store.darkMode && 'bg-black'"> </div>`, set empty x-data, and then get it from $store. As for why your method works, I guess it is that after the livewire operation, the original object cannot be used normally and must be retrieved from $store. In addition, I want to say that it is a bit cumbersome to start fetching from the root object `$store` every time. Finally, thank you very much for your help.

**Comment by xiaoxidashen** [comment]
Let me summarize, use livewire components (I use volt) instead of blade anonymous templates. Insert js code between @script and @endscript. The location inserted into the layout component should be in the body instead of the head. When `Alpine.store` does not need to be written in `document.addEventListener('alpine:init', () => {});` when it is between @script and @endscript 
Thanks to @danie-ramdhani for all your help
